File: backend/turip-app/src/main/java/turip/content/controller/dto/response/content/ContentResponse.java
package turip.content.controller.dto.response.content;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDate;
import turip.content.domain.Content;
import turip.creator.controller.dto.response.CreatorResponse;
import turip.region.controller.dto.response.CityNameResponse;

public record ContentResponse(
        Long id,
        String title,
        String url,
        LocalDate uploadedDate,
        CityNameResponse city,
        CreatorResponse creator,
        @JsonProperty("isBookmarked")
        boolean isFavorite
) {
    public static ContentResponse of(Content content, boolean isFavorite) {
        return new ContentResponse(
                content.getId(),
                content.getTitle(),
                content.getUrl(),
                content.getUploadedDate(),
                CityNameResponse.from(content.getCity()),
                CreatorResponse.from(content.getCreator()),
                isFavorite
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/content/dto/ContentResponse.kt
package com.on.turip.data.content.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ContentResponse(
    @SerialName("creator")
    val creator: CreatorInformationResponse,
    @SerialName("id")
    val id: Long,
    @SerialName("title")
    val title: String,
    @SerialName("uploadedDate")
    val uploadedDate: String,
    @SerialName("url")
    val url: String,
    @SerialName("city")
    val city: CityResponse,
    @SerialName("isFavorite")
    val isFavorite: Boolean = false, // TODO 추후에 서버에서 값이 내려오면 삭제
)
